### PR TITLE
Don't create default SSLContext if ssl module isn't present

### PR DESCRIPTION
**Pull request by nateprewitt**
This PR is to address a [recent regression in Emscripten support for urllib3](https://github.com/urllib3/urllib3/actions/runs/9253144050/job/25452248794). We began unilaterally creating an SSLContext without consideration for Python versions built without an ssl module. This is handled in urllib3 but missed in our usage of `create_urllib3_context`.

This PR expands on the fix in #6716 by also evaluating both the presence of the `ssl` module before creating and setting the default SSLContext.

### CODE REVIEW

File: src/requests/adapters.py [hunk 1]
```diff
@@ -90,13 +98,19 @@ def _urllib3_request_context(
...
     has_poolmanager_ssl_context = poolmanager_kwargs.get("ssl_context")
+    should_use_default_ssl_context = (
+        _preloaded_ssl_context is not None and not has_poolmanager_ssl_context
+    )
 
     cert_reqs = "CERT_REQUIRED"
     if verify is False:
         cert_reqs = "CERT_NONE"
-    elif verify is True and not has_poolmanager_ssl_context:
+    elif verify is True and should_use_default_ssl_context:
```
Comment (sigmavirus24) [inline src/requests/adapters.py:113]: Missed this earlier but do we need any logic for verify=True without a condition here?

Comment (nateprewitt) [reply src/requests/adapters.py:113]: Are you asking about the behavior if `verify is True` but `should_use_default_ssl_context` is False? This was a no-op prior to #6655 which fellback to letting the SSLContext be created when fetching the connection from the pool. [`_ssl_wrap_socket_and_match_hostname`](https://github.com/urllib3/urllib3/blob/b07a669bd970d69847801148286b726f0570b625/src/urllib3/connection.py#L755-L765) handles this for us, so I don't know if there's any additional concern from our previous behavior unless I'm missing part of the question.

Comment (sigmavirus24) [reply src/requests/adapters.py:113]: Makes sense. I've a terrible headache so just looking at the branches and concerned about missing something is all.

**Review by sigmavirus24 — APPROVED** [review]

File: src/requests/adapters.py (hunk 1 above)
Comment (nateprewitt) [reply src/requests/adapters.py:113]: Nope, that's a good call out. I'll give it one more look before merging but I think we're ok for that case.

Comment (sigmavirus24) [reply src/requests/adapters.py:113]: Oh, looking again, not on a phone, and without a migraine, I see the `cert_reqs = "CERT_REQUIRED"` line on L110 that I clearly wrote, and which is what I was thinking we may want to be concerned about. So, I'm not nearly as worried.

File: src/requests/adapters.py [hunk 2]
```diff
@@ -90,13 +98,19 @@ def _urllib3_request_context(
...
         pool_kwargs["ssl_context"] = _preloaded_ssl_context
     elif isinstance(verify, str):
         if not os.path.isdir(verify):
```
Comment (sigmavirus24) [inline src/requests/adapters.py:116]: Looking at this again with something of a performance mindset, do we want to cache somehow lookups to `isdir`? I'm not sure this hurts us at all, but just thinking about things that could slow us down in certain cases now. (Not for this pull request, just putting it out there)

**Review by sigmavirus24 — APPROVED** [review]

File: src/requests/adapters.py (hunk 2 above)
Comment (nateprewitt) [reply src/requests/adapters.py:116]: That seems like a reasonable optimization, I guess we'd need to check how much time we're actually spending on the dir check. I assume we'll get feedback if we have cases where this is a bottleneck.

**Comment by AdamWill** [comment]
We also need to do this for `FileNotFoundError`, which is raised if `DEFAULT_CA_BUNDLE_PATH` isn't there. Sent as https://github.com/psf/requests/pull/6781 .

**Review by Pjrich1313 — COMMENTED** [review]
Rerun have timestamp

**Pull request merged**